materialbarcodescanner/src/main/java/com/edwardvanraak/materialbarcodescanner/MaterialBarcodeScannerFragment.java

```diff
@@ -112,10 +124,27 @@ public void onDestroy() {
         clean();
     }
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode != MaterialBarcodeScanner.RC_HANDLE_CAMERA_PERM) {
+            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        }
+        else if (grantResults.length != 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            EventBus.getDefault().register(this);
+        }
+        else {
+            displayCameraPermissionDeniedMessage();
+        }
+    }
+
     protected void setNewDetectionListener(NewDetectionListener newDetectionListener) {
         this.newDetectionListener = newDetectionListener;
     }
 
+    protected void displayCameraPermissionDeniedMessage() {
+        Toast.makeText(getActivity(), "Camera permission required to scan codes!", Toast.LENGTH_LONG).show();
```

Comment:
let's move this to strings.xml

Reply:
True... I did not move it there initially because the intent was to always override this method. But let's not break conventions anyway :)